besu/src/main/java/org/hyperledger/besu/components/BesuComponent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.hyperledger.besu.components;
+
+import org.hyperledger.besu.cli.BesuCommand;
+import org.hyperledger.besu.ethereum.bonsai.cache.CachedMerkleTrieLoader;
+import org.hyperledger.besu.ethereum.bonsai.cache.CachedMerkleTrieLoaderModule;
+import org.hyperledger.besu.metrics.MetricsSystemModule;
+import org.hyperledger.besu.metrics.ObservableMetricsSystem;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Component;
+import org.slf4j.Logger;
+
+/** An application context that knows how to provide dependencies based on Dagger setup. */
+@Singleton
+@Component(
+    modules = {
+      BesuCommandModule.class,
+      MetricsSystemModule.class,
+      CachedMerkleTrieLoaderModule.class
+    })
+public interface BesuComponent {
```

Comment:
BesuComponent seems quite generic to me. Could/should we be more precise with the name? Or does it refer to the one and only component for the `besu` gradle module? 
Will this end up being _the_ application context, or one of many (e.g. one per module)?

Reply:
Any reason not to call the whole class ApplicationContext? Or is that to Spring-y 😛

Reply:
Yes, trying to adopt Daggers convention